src/components/container/HeartEssentialEight/HeartEssentialEight.tsx

```diff
@@ -0,0 +1,215 @@
+      newScore = (newScore / 800) * 100;
+    }
+
+    setScore(parseFloat(newScore.toFixed(1)));
+    setAchievementMap(newAchievementMap);
+  }
+
+  function generateAchievementMap(
+    didAchieve: boolean,
+    newAchievementMap: Record<HeartEssentialEightScores, boolean>
+  ) {
+    const map = newAchievementMap as Record<string, boolean>;
+    const achievementList: any[] = [];
+    for (const key in map) {
+      const vv = map[key];
+      if (vv === didAchieve) {
+        const title = key.replace(/([A-Z])/g, " $1").replace(/^./, (str) => {
+          return str.toUpperCase();
+        });
+        let icon: any;
+        let subtitle: any;
+        let url: string;
+        const circleClasses: string[] = ["circle"];
+        switch (key) {
+          case "Activity":
+            icon = iconGetActive;
+            subtitle="Display Active Time from Fitbit / Garmin.  Probably AppleExerciseTime for Apple Health. Or manual entry";
+            circleClasses.push("activity");
+            break;
+          case "BloodPressure":
+            icon = iconManageBloodPressure;
+            subtitle="Display most recent value from EHR, or manual entry";
+            circleClasses.push("bloodPressure");
+            break;
+          case "BloodSugar":
+            icon = iconReduceBloodSugar;
+            subtitle="Display Glucose numbers from Dexcom, or manual entry";
+            circleClasses.push("bloodSugar");
+            break;
+          case "BMI":
+            icon = iconLoseWeight;
+            subtitle="Display BMI calculated from survey data"
+            circleClasses.push("bmi");
+            break;
+          case "Cholesterol":
+            icon = iconCholesterol;
+            subtitle="Display most recent value from EHR, or manual entry";
+            circleClasses.push("cholesterol");
+            break;
+          case "Diet":
+            icon = iconEatWell;
+            subtitle="Eat more whole foods / Cut back on candy / Whatever the key points are from the survey";
+            circleClasses.push("diet");
+            break;
+          case "Nicotine":
+            icon = iconStopSmoking;
+            subtitle="Stop smoking / Congratulations on { 5+ years/never smoking }"
+            circleClasses.push("nicotine");
+            break;
+          case "Sleep":
+            icon = iconSleep;
+            subtitle="Display value from Apple Health/Fitbit/Garmin, or manual entry"
+            circleClasses.push("sleep");
+            break;
+          default:
+            icon = null;
+        }
+
+        const achievementIcon = (
+          <div className={circleClasses.join(" ")}>
+            <img src={icon} />
+          </div>
+        );
+
+        const achievement = (
+            <Action
+              key={`achievement-${key}`}
+              title={title}
+              subtitle={subtitle}
+              icon={achievementIcon}
+              onClick={() => MyDataHelps.openExternalUrl(url)}
```

Comment:
We may want to have this call a handler function on the props.  That way, when this is added to the view builder, it can use an `ActionDefinition` setting to allow the person building the view to configure what happens when each of the types is clicked.  Maybe it opens a URL, launches another app, or goes to a different tab, etc.

If we go that route, we'd probably want to support configurable subtitle text for each action as well.